web/react/components/error_bar.jsx

```diff
@@ -31,6 +61,14 @@ export default class ErrorBar extends React.Component {
         ErrorStore.removeChangeListener(this.onErrorChange);
     }
 
+    componentWillUpdate(nextProps, nextState) {
+        if (this.isValidError(nextState)) {
```

Comment:
Actually I guess it doesn't matter because we are not modifying ourselves, we are modifying another component. (YUCK...)  Actually that means this can go in render. 0/5 because it sucks anyways.